### PR TITLE
ceph-disk: test activate on the second journal

### PR DESCRIPTION
**Pull request by dachary**
This series has been manually tested on Ubuntu 14.04, CentOS 7, RHEL 7, CentOS 6.5 and RHEL 6.5 on virtual machines and a native distribution installed, with a compilation from sources and the following command:
<pre>
sudo env PATH=:$PATH test/ceph-disk.sh test_activate_dev
</pre>
run from the src directory. In the case of CentOS 6.5 and RHEL 6.5 an additional reboot was necessary to add loop.max_part=16 to the kernel cmdline.

### CODE REVIEW

**Comment by dachary** [comment]
https://github.com/dachary/ceph/commit/b0a7e725f5bcfcbcdc1bf2424979522b2de2c0eb succeeds on ubuntu 14.04 with a 3.13 kernel.

**Comment by dachary** [comment]
@alfredodeza this is in the context of http://tracker.ceph.com/issues/10987 . It can be used to conveniently show that the intended use case actually works or fail without manual and volatile operations. It's one step closer to have proper integration tests for ceph-disk which is a nice plus ;-)

File: src/ceph-disk [hunk 1]
```diff
@@ -988,31 +988,20 @@ def get_free_partition_index(dev):
 
     if not lines:
         raise Error('parted failed to output anything')
-    lines = str(lines).splitlines(True)
-
-    # work around buggy libreadline(?) library in rhel/centos.
-    idiot_prefix = '\x1b\x5b\x3f\x31\x30\x33\x34\x68'
```
Comment (alfredodeza) [inline src/ceph-disk]: why is this not needed anymore?

File: src/ceph-disk [hunk 2]
```diff
@@ -988,31 +988,20 @@ def get_free_partition_index(dev):
...
-    num = 1
-    while num in seen:
-        num += 1
-    return num
+    if ('CHS;' not in lines and
+        'CYL;' not in lines and
+        'BYT;' not in lines):
+        raise Error('parted output expected to contain one of ' +
+                    'CHH; CYL; or BYT; : ' + lines)
+    if dev not in lines:
```
Comment (alfredodeza) [inline src/ceph-disk]: why just check this vs what was done before? `if not lines[0].startswith('/dev/')` 

Is it ok if it is not at the start of the string?

File: src/ceph-disk [hunk 3]
```diff
@@ -988,31 +988,20 @@ def get_free_partition_index(dev):
...
+        raise Error('parted output expected to contain ' + dev + ': ' + lines)
+    _, partitions = lines.split(dev)
+    partition_numbers = map(lambda p: int(p),
```
Comment (alfredodeza) [inline src/ceph-disk]: this is a bit hard to follow, it would be nice to extract it as a utility that in turn can be unit tested, which in turn may not
need to use all these built-ins with `re` in two lines

File: src/ceph-disk (hunk 1 above)
Comment (dachary) [inline src/ceph-disk]: because we change the sanity check strategy to be looser

File: src/ceph-disk (hunk 2 above)
Comment (dachary) [inline src/ceph-disk]: the previous code assumes the format of parted is rigid and does not vary accross implementations. It turns out to not be the case. We could either task ourselves with the impossible mission of matching rigidly the format of all parted implementation. Or chose to implement a few sanity checks that gives us a reasonable certainty that the output is indeed something that looks like what parted would output.

File: src/ceph-disk (hunk 3 above)
Comment (dachary) [inline src/ceph-disk]: ceph-disk.sh contains integration tests that would break if this failed (i.e. the integration tests rely on this function returning 2 when there is 1 partition already).

**Comment by dachary** [comment]
sudo test/ceph-disk.sh test_activate_dev passes

**Comment by alfredodeza** [comment]
this looks to me
